src/vma/dev/cq_mgr.cpp

```diff
@@ -853,6 +853,34 @@ int cq_mgr::drain_and_proccess(uintptr_t* p_recycle_buffers_last_wr_id /*=NULL*/
 	return ret_total;
 }
 
+//  1 -> busy
+//  0 -> ok
+// -1 -> error
+int cq_mgr::ack_and_request_notification()
+{
+	int res, cq_ev_count = 0;
+	ibv_cq* ib_cq;
+	void *cq_context;
+	do {
+		res = ibv_get_cq_event(m_comp_event_channel, &ib_cq, &cq_context);
+		if (res == 0) {
+			++cq_ev_count;
+		}
+	} while (res == 0);
+	if (errno != EAGAIN) {
+		return -1;
+	}
+	if (cq_ev_count > 0) {
+		ibv_ack_cq_events(m_p_ibv_cq, cq_ev_count);
```

Comment:
Probably calling `get_cq_event()` `cq_ev_count` times is needed as were done
See https://github.com/Mellanox/libvma/blob/master/src/vma/dev/cq_mgr.cpp#L945-L951
or
https://github.com/Mellanox/libvma/blob/master/src/vma/dev/cq_mgr.cpp#L904-L917

RM: https://github.com/Mellanox/libvma/commit/9c3b43e05797cfc5e66d03dfb41fb6b5b6c30298#diff-2170492bb09a82a83f97f7f158592e7c

Reply:
void vma_ib_mlx5_get_cq_event(vma_ib_mlx5_cq_t *)
{
	// no need in operation with cq_sn as far as it is managed by driver code for now
}

Reply:
it doe's nothing.
is there a test for the event?

Reply:
done

Reply:
get_cq_event() just increment to 1. In your case you should call get_cq_event() cq_ev_count times.

Reply:
done